lib/client/db/profile.go

```diff
@@ -141,9 +141,9 @@ func IsSupported(db tlsca.RouteToDatabase) bool {
 func load(tc *client.TeleportClient, db tlsca.RouteToDatabase) (profile.ConnectProfileFile, error) {
 	switch db.Protocol {
 	case defaults.ProtocolPostgres:
-		return postgres.Load(tc.HomePath)
+		return postgres.Load(tc.MockUserHomeDir)
```

Comment:
Definitely - I think calling this Mock here is actually very misleading. Treat this as any other dependency.

Reply:
I renamed it to `MySQLOptionFilePathOverride` and `PostgresServiceFilePathOverride` since it's only specific to those files in test and I don't know of anything else that needs to override the home dir.

[4cf47ef](https://github.com/gravitational/teleport/pull/29220/commits/4cf47ef0faf998c4ffb9c707aec1990f21c5cfdf)